cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexCreator.java

```diff
@@ -394,4 +389,10 @@ private static String generateIndexName(Set<String> existingIndexNames) throws E
         }
     }
 
+    private static class ListIndexesCallable implements SQLCallable<List<Index>> {
```

Comment:
Right it is a reduction in memory overhead. We can be pragamatic, if `IndexCreator` is a singleton then I think we're only ever going to have one copy of it and it's anonymous inner. I think the warning is more relevant for cases where we might generate hundreds of instances of inners each holding an unnecessary reference (in the case that they could be a replaced by a static class). Nonetheless we have a made a decision to extract callables so we should probably just be consistent.